### PR TITLE
ARROW-8705: [Java] copying null values from ComplexCopier

### PR DESCRIPTION
**Pull request by tvamsikalyan**

### CODE REVIEW

File: java/vector/src/main/codegen/templates/UnionListWriter.java [hunk 1]
```diff
@@ -180,7 +180,7 @@ public void write(DecimalHolder holder) {
 
   @Override
   public void writeNull() {
-    writer.setPosition(writer.idx()+1);
```
Comment (pravindra) [inline java/vector/src/main/codegen/templates/UnionListWriter.java:183]: can you please add a check here 

if (writer.idx() >= (idx() + 1) * listSize) {
      throw new IllegalStateException(String.format("values at index %s is greater than listSize %s", idx(), listSize));
    }

File: java/vector/src/main/codegen/templates/StructWriters.java [hunk 2]
```diff
@@ -196,6 +196,15 @@ public void setPosition(int index) {
     }
   }
 
+  <#if mode="Nullable">
+  @Override
+  public void writeNull() {
+    container.setNull(idx());
+    setValueCount(idx()+1);
```
Comment (pravindra) [inline java/vector/src/main/codegen/templates/StructWriters.java:203]: is the setValueCount required ?  it should be done after all mutations to the vector are done.

**Review by pravindra — COMMENTED** [review]
- please add a description the jira about the bug (what/when)
- please add a description in the PR about the fix

File: java/vector/src/main/codegen/templates/UnionFixedSizeListWriter.java [hunk 3]
```diff
@@ -183,7 +183,7 @@ public void write(DecimalHolder holder) {
 
   @Override
   public void writeNull() {
-    writer.setPosition(writer.idx()+1);
+    writer.writeNull();
```
Comment (pravindra) [inline java/vector/src/main/codegen/templates/UnionFixedSizeListWriter.java:186]: can you please add a check here

if (writer.idx() >= (idx() + 1) * listSize) {
throw new IllegalStateException(String.format("values at index %s is greater than listSize %s", idx(), listSize));
}

File: java/vector/src/main/codegen/templates/UnionListWriter.java (hunk 1 above)
Comment (pravindra) [reply java/vector/src/main/codegen/templates/UnionListWriter.java:183]: sorry, wrong place.

File: java/vector/src/main/codegen/templates/StructWriters.java (hunk 2 above)
Comment (tvamsikalyan) [reply java/vector/src/main/codegen/templates/StructWriters.java:203]: same code is present in the other writers too, so ignoring this comment

File: java/vector/src/main/codegen/templates/UnionFixedSizeListWriter.java (hunk 3 above)
Comment (tvamsikalyan) [reply java/vector/src/main/codegen/templates/UnionFixedSizeListWriter.java:186]: done

**Review by pravindra — APPROVED** [review]

**Review by praveenbingo — APPROVED** [review]
looks good to me.

**Comment by wesm** [comment]
Remember to include the component in the PR title

**Pull request closed without merging**